src/channel.cr

```diff
@@ -96,13 +96,21 @@ class Channel(T)
       @closed = true
 
       @senders.each do |sender|
-        sender.state_ptr.value = DeliveryState::Closed
-        sender.fiber.enqueue
+        if (select_context = sender.select_context) && !select_context.try_trigger
+          next
```

Comment:
As `next` would exit the current block, this could be rewritten to a guard like row, ie 
```
next if (select_context = sender.select_context) && !select_context.try_trigger

sender.state_ptr.value = DeliveryState::Closed
sender.fiber.enqueue
```

Reply:
I prefer to use explicit `if-else` rather than one line. The explicit way can indicate which fiber is allow to be enqueued into the scheduler. And this line `next if (select_context = sender.select_context) && !select_context.try_trigger` is too long for me.